gattc_sd.go

```diff
@@ -56,8 +67,8 @@ func (d *Device) DiscoverServices(uuids []UUID) ([]DeviceService, error) {
 		return nil, errAlreadyDiscovering
 	}
 
-	services := make([]DeviceService, len(uuids))
-	for i, uuid := range uuids {
+	services := []DeviceService{}
```

Comment:
This change may introduce additional heap allocations. By allocating the whole slice at the start and setting individual locations in the slice, only a single allocation is done of exactly the right size.

If you really want to use `append`, you can do the following:

```suggestion
	services := make([]DeviceService, 0, len(uuids))
```

This sets the capacity while `len(services)` remains at 0. `append` will probably still be slightly more expensive than just setting a field but it won't cause any additional heap allocations. However, I would prefer to just keep it as-is.

Reply:
I am trying to lay the groundwork for eliminating the need to know all of the specific service/characteristic UUIDs in order to use `DiscoverServices()` and `DiscoverCharacteristics()`.  How else can we do that, set a max number as limit?

Reply:
Chose to use reasonable numbers as default limits.